File: src/PythonBridge.py
from storage import Storage
from networking import JiraClient

class PythonBridge:
    def __init__(self):
        self.storage = Storage()
        self.jira = None

    def init_jira(self, base_url, username, api_key):
        self.jira = JiraClient(base_url, username, api_key)

    def save_entry(self, entry):
        self.storage.add_entry(entry)

    def get_entries(self):
        return self.storage.get_entries()

    def get_projects(self):
        if self.jira:
            return self.jira.get_projects()
        return []

File: src/PythonBridge.swift
import Foundation
import PythonKit

@MainActor
struct PythonBridge {
    static func saveEntry(duration: TimeInterval) {
        let py = Python("python")
        let storage = py.import("src.storage")
        let store = storage.Storage()
        store.addEntry(project: "DevProject", duration: duration, notes: "")
    }
    
    static func saveSettings(base_url: String, username: String, api_key: String) {
        let py = Python("python")
        let storage = py.import("src.storage")
        let store = storage.Storage()
        store.updateSettings(base_url: base_url, username: username, api_key: api_key)
    }
    
    static func loadProjects(completion: @escaping ([String]) -> Void) {
        let py = Python("python")
        let storage = py.import("src.storage")
        let store = storage.Storage()
        let entries = store.getEntries()
        let names: [String] = entries.map { $["project"] as! String }
        completion(names)
    }
}
